src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -690,6 +706,27 @@ static int client_append_oro(sd_dhcp6_client *client, uint8_t **buf, size_t *off
         return dhcp6_option_append(buf, offset, SD_DHCP6_OPTION_ORO, n * sizeof(be16_t), req_opts);
 }
 
+static int client_append_mudurl(sd_dhcp6_client *client, uint8_t **buf, size_t *offset) {
+        int r;
+
+        assert(client);
+        assert(buf);
+        assert(*buf);
+        assert(offset);
+
+        if (!client->mudurl)
+                return 0;
+
+        if (client->state != DHCP6_STATE_STOPPING) {
```

Comment:
As @poettering suggested, please return earlier:
```c
if (client->state == DHCP6_STATE_STOPPING)
        return 0;

r = dhcp6_option_append(...);
```

Reply:
yes fixed in the last push